File: problems/problems_2576/solution.go
package problem2576

import (
	"encoding/json"
	"log"
	"sort"
	"strings"
)

func maxNumOfMarkedIndices(nums []int) int {
	sort.Ints(nums)
	left := 0
	for right := (len(nums) + 1) / 2; right < len(nums); right++ {
		if nums[right] >= 2*nums[left] {
			left++
		}
	}
	return left << 1
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return maxNumOfMarkedIndices(nums)
}

File: problems/problems_2576/solution.ts
function maxNumOfMarkedIndices(nums: number[]): number {
	nums.sort((a, b) => a - b);
	let left: number = 0;
	const n: number = nums.length;
	for (let right: number = Math.floor((n + 1) / 2); right < n; right++) {
		if (nums[right] >= 2 * nums[left]) {
			left++;
		}
	}
	return left * 2;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return maxNumOfMarkedIndices(nums);
}
